### ISSUE TITLE
Optional fields & arguments

### ISSUE THREAD

**Issue by MartinKavik** [trevorblades/countries#14]
Hi!
Is there a reason why all fields and arguments are optional?

I'm writing GraphQL example for [Seed](https://github.com/seed-rs/seed) and it makes code unnecessary complex because we have to take into account all valid possibilities.

Thank you.

**Comment by trevorblades** [comment]
@MartinKavik good call. I just made a commit that makes the `code` argument required for singular field resolvers (ie. `country(code: "CA")`). Could you explain more about what you mean by "required fields"?

**Comment by MartinKavik** [comment]
Something like that:
```graphql
  type Country {
    code: String!
    ...
    languages: [Language!]!   # there is always at least one language
    states: [State!]          # null or at least one state
  }
```

---

Also I would expect that 
  - `language.rtl` is `Boolean!`
  - `code` is `ID!` (?)
  - all entities are sorted / can I sort them somehow (?)

(I'm not sure if my suggestions make sense because I don't have much experience with GraphQL, please correct me.)
Complete example in Seed - https://github.com/seed-rs/seed/issues/400#issuecomment-604286864

**Comment by trevorblades** [comment]
I made a PR that introduces nullability into the schema, clearly indicating what fields can and can't be null. I've also cast `language.rtl` as a `Boolean!` as you suggested.

Also, rather than making `code` an `ID!`, I made each of the codes used in field arguments an enum of only the valid codes.

Would love to hear your thoughts on these changes!

**Comment by MartinKavik** [comment]
I don't know if `enum` for `code`s is a good idea:
- They are inconsistent among entities:
  - `Continent` => `ContinentCode` (upper-case)
  - `Country` => `CountryCode` (upper-case)
  - `State` => optional non-unique `String` (upper-case)
  - `Language` => `LanguageCode` (lower-case)
- It makes schema less readable (see `SCHEMA` on the playground).
- We would need to recompile server and all clients once some people win the fight for independence and establish a new country...

**Comment by MartinKavik** [comment]
I've cloned your PR and adapt my Seed example to it. Works good,  I just added `capital` field and a link to that example - PR https://github.com/trevorblades/countries/pull/16.

**Comment by trevorblades** [comment]
Yeah, good points re: enums. I liked them because it made it clear what the accepted values were. We would need to recompile the server every time a new country appeared or an old one changed either way, since the data is coming from `countries-list`, so we'd upgrade the dependency and restart.

The question for me is: "Is this change good enough to warrant breaking existing uses of this API?" to which it sounds like the answer is no.

I've changed the `code` fields in the PR from enums to `ID!`. Regarding the inconsistency of the nullable `String` field on the `State` type, this is unfortunately the reality of the dataset that I've got, and the fact that not every state in the world has a short code associated with it. This is also why states aren't independently queryable in this API.
